crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -271,35 +273,37 @@ where
             flox_nix_content_with_new_packages(&current_manifest_contents, packages)?;
         match new_manifest_contents {
             ManifestContent::Unchanged => return Ok(false),
-            ManifestContent::Changed(new_manifest_contents) => {
-                let mut temp_env = self.make_temporary()?;
-                temp_env.update_manifest(&new_manifest_contents)?;
-                temp_env.build(nix, system).await?;
-                self.replace_with(temp_env)?;
+            ManifestContent::Changed(contents) => {
+                self.transact_with_manifest_contents(contents, nix, system)
+                    .await?;
```

Comment:
Is there a more idiomatic way to map this to `Ok(true)`?

Reply:
Added a suggestion below, but i think this is idiomatic enough..